docs-website/topics/serialization-json-configuration.md

```diff
@@ -0,0 +1,707 @@
+[//]: # (title: Customize the Json instance)
+
+The default [`Json`](https://kotlinlang.org/api/kotlinx.serialization/kotlinx-serialization-json/kotlinx.serialization.json/-json/) class enforces Kotlin type safety and only allows values that can be serialized into standard JSON.
+
+You can handle more flexible or non-standard JSON features by creating a custom `Json` instance with the [`Json()`](https://kotlinlang.org/api/kotlinx.serialization/kotlinx-serialization-json/kotlinx.serialization.json/-json.html) builder function:
+
+```kotlin
+// Creates a Json instance based on the default configuration, ignoring unknown keys
+val customJson = Json {
+    ignoreUnknownKeys = true
+}
+
+// Use the customJson instance with the same syntax as the default one to encode a string
+val jsonString = customJson.encodeToString(user)
+println(jsonString)
+```
+
+`Json` instances created this way are immutable and thread-safe, which makes them safe to store in a top-level property for reuse.
+
+You can also base a new `Json` instance on an existing one and modify its settings with the [`JsonBuilder`](https://kotlinlang.org/api/kotlinx.serialization/kotlinx-serialization-json/kotlinx.serialization.json/-json-builder/) DSL:
+
+```kotlin
+// Creates a new instance based on an existing Json
+val lenientJson = Json(customJson) {
+    isLenient = true
+    prettyPrint = true
+}
+```
+
+> Reusing custom `Json` instances improves performance by allowing them to cache class-specific information.
+>
+{style="tip"}
+
+The following sections cover the various `Json` class configuration features.
+
+## Customize JSON serialization
```

Comment:
I feel that the separation between "Customize JSON serialization" and "Customize JSON deserialization" is rather confusing, since a lot of settings, such as `allowSpecialFloatingPointValues`, `explicitNulls`, `allowStructuredMapKeys`, etc. affect both of them. Or we could separate them into "General settings" or smth

Reply:
As I said, configuration flags sometimes affect both input and output. So we either have to have three categories ("Customize Both", "Customize Serialization output", "Customize deserialization behavior") or invent something else